server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -317,19 +325,82 @@ public IndexInput openInput(String name, IOContext context) throws IOException {
         }
     }
 
-    public void copyFrom(Directory from, String src, String dest, IOContext context, boolean useCommonSuffix) throws IOException {
+    /**
+     * This acquires a lock on a given commit by creating a lock file in lock directory using {@code FileLockInfo}
+     * @param primaryTerm Primary Term of index at the time of commit.
+     * @param generation Commit Generation
+     * @param acquirerId Lock Acquirer ID which wants to acquire lock on the commit.
+     * @throws IOException if the file for which we are acquiring lock is not found
+     */
+    @Override
+    public void acquireLock(long primaryTerm, long generation, String acquirerId) throws IOException {
+        String metadataFile = getMetadataFileForCommit(primaryTerm, generation);
+
+        mdLockManager.acquire(FileLockInfo.getLockInfoBuilder().withFileToLock(metadataFile).withAcquirerId(acquirerId).build());
+    }
+
+    /**
+     * Releases a lock which was acquired on given segment commit.
+     * @param primaryTerm Primary Term of index at the time of commit.
+     * @param generation Commit Generation
+     * @param acquirerId Acquirer ID for which lock needs to be released.
+     * @throws IOException in case of any failure during releasing lock.
+     */
+    @Override
+    public void releaseLock(long primaryTerm, long generation, String acquirerId) throws IOException {
+        String metadataFile = getMetadataFileForCommit(primaryTerm, generation);
+        mdLockManager.release(FileLockInfo.getLockInfoBuilder().withFileToLock(metadataFile).withAcquirerId(acquirerId).build());
+    }
+
+    /**
+     * Checks if a specific commit have any corresponding lock file.
+     * @param primaryTerm Primary Term of index at the time of commit.
+     * @param generation Commit Generation
+     * @return True if there is at least one lock for given primary term and generation.
+     * @throws IOException in case of any failure during check for lock.
+     */
+    @Override
+    public Boolean isLockAcquired(long primaryTerm, long generation) throws IOException {
+        String metadataFile = getMetadataFileForCommit(primaryTerm, generation);
```

Comment:
as discussed, it should throw NoSuchFileException, if it don't have the metadata file corresponding to a PrimaryTerm and Generation. otherwise if we return true, caller may remain in the assumption that metadata file still persist but it do not have any locks.